invenio_rdm_migrator/streams/actions/files.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 CERN.
+#
+# Invenio-RDM-Migrator is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Files actions module."""
+
+from dataclasses import dataclass
+
+from ...actions import LoadAction, LoadData
+from ...load.postgresql.transactions.operations import Operation, OperationType
+from ..models.files import FilesBucket, FilesInstance, FilesObjectVersion
+
+
+@dataclass
+class FileUploadData(LoadData):
+    """File upload action data."""
+
+    bucket: dict
+    object_version: dict
+    file: dict
```

Comment:
minor: shouldn't we actually be targeting also the "final" RDM record models, i.e. `RDMRecordFile`, `RDMDraftFile`, etc.? I know this complicates the task significantly since you need to be able to resolve the record/draft from the bucket (using the state I assume).

Reply:
What kind of "target" would you have in mind? If it is the `bucket`, this code assumes that it exists since it should have happened in a previous transaction (`DraftCreateAction`), and the `bucket_id` value is kept (we do not modify the UUID).

On the one hand I see it more safe to use the state to assert this, on the other hand we fall again in the fact that we have not been using the state for this. It's the same discussion we had the other day for the user actions, for example _do we check if it exists before running a deactivation_, etc.

WDYT?

Reply:
Discussed in Discord https://discord.com/channels/692989811736182844/1118079987891843093/1139140395285098587

Reply:
I mean that in legacy, for file upload action it's enough to just update the `Bucket`, and create `FileInstance` and `ObjectVersion` rows, that's all the tables/models that need changes.

In InvenioRDM, for a file upload you need to do all of the above and additionally also create an `RDMRecordFile`/`RDMDraftFile` row.

Reply:
Implemented, had to add to the table generator (snapshot migration) the bucket state interaction for those cases where the draft might come in the initial migration but files later on. On the other hand, I'm wondering if we should have some sort of "initialization" for the state. Meaning, when we start the actions migration we load from db all that's needed from state (I wonder if we actually have this information for all the cases).

Reply:
Yeah, we might have to backtrack a bit on the snapshot pipeline to see if we keep all the information we need in state. The good thing is that we'll catch all these cases, since if we see information missing while implementing the actions there's only one place we can get it 😅